src/components/Timer.js

```diff
@@ -1,36 +1,33 @@
 import React from 'react';
+import TimerDisplay from "./TimerDisplay"
+import TimeControlButtons from "./TimerControlButtons"
 
 class Timer extends React.Component{
     constructor(){
         super();
-        this.state = {beginTime: new Date(), endTime: new Date(), timeElapsed: 0};
-
-        this.resetClock = this.resetClock.bind(this);
-        this.incrementClock = this.incrementClock.bind(this);
-        this.startClock = this.startClock.bind(this);
-        this.stopClock = this.stopClock.bind(this);
-
-        this.isStarted = false;
+        this.state = {beginTime: new Date(), endTime: new Date(), timeElapsed: 0, isStarted: false};
     }
 
     componentWillUnmount(){
         this.resetClock();
     }
 
-    startClock(){
+    startClock = () =>{
         this.setState(state => ({
             beginTime: new Date(),
         }))
         this.intervalId = setInterval(this.incrementClock, 1000);
         this.isStarted = true;
     }
 
-    stopClock(){
+    stopClock = () => {
         clearInterval(this.intervalId);
-        this.isStarted = false;
+        this.setState(state => ({
+            isStarted: true,
+        }))
     }
 
-    resetClock(){
+    resetClock = () => {
         this.setState(state => ({
             beginTime: new Date(), 
```

Comment:
Przy starcie ustawiasz `beginTime` na `new Date()`, przy resecie robisz to ponownie. Wyglada na zbędną linie, mały szczegół ale rzuca się w oczy.